bc_obps/reporting/service/report_product_service.py

```diff
@@ -59,24 +57,25 @@ def save_production_data(
                 },
             )
 
-        # Add reportProduct for Fat, Oil & Grease unregulated product for emission allocation only
-        if fog_product_id in ReportOperation.objects.get(
-            report_version_id=report_version_id
-        ).regulated_products.values_list("id", flat=True):
-            ReportProduct.objects.update_or_create(
-                report_version_id=report_version_id,
-                facility_report=facility_report,
-                product_id=fog_product_id,
-                defaults={
-                    "report_version_id": report_version_id,
-                    "facility_report": facility_report,
-                    "product_id": fog_product_id,
-                    "annual_production": 0,
-                    "production_data_apr_dec": 0,
-                    "production_methodology": "other",
-                    "production_methodology_description": "auto-generated report_product record for unregulated product",
-                },
-            )
+        # Add a report_product record for unregulated products reported by this operation (for emission allocation only)
+        for unregulated_product_id in unregulated_product_ids:
+            if unregulated_product_id in ReportOperation.objects.get(
+                report_version_id=report_version_id
+            ).regulated_products.values_list("id", flat=True):
+                ReportProduct.objects.update_or_create(
+                    report_version_id=report_version_id,
```

Comment:
- we might want to query the list of product ids outside of the for loop to avoid multiple database hits
- this means that if an unregulated product is removed from the list on the report operation page, we'll still have a dangling record, we should remove it earlier with the other ones on line 39.

I think the easy thing is to just use the list of reported unregulated products (Line 20) every time, instead of querying the system for all unregulated products and filtering later whether they're to be reported